src/cpp/CMakeLists.txt

```diff
@@ -49,10 +49,19 @@ elseif(NOT EPROSIMA_INSTALLER)
     # Create library
     add_library(${PROJECT_NAME} ${${PROJECT_NAME}_SOURCES})
     set_common_compile_options(${PROJECT_NAME})
-    set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${PROJECT_VERSION})
-    set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR})
+    set_target_properties(${PROJECT_NAME} PROPERTIES
+        VERSION ${PROJECT_VERSION}
+        SOVERSION ${PROJECT_VERSION_MAJOR}
+        )
 
-    target_compile_definitions(${PROJECT_NAME} PRIVATE ${PROJECT_NAME_UPPER}_SOURCE)
+    target_compile_definitions(${PROJECT_NAME}
+        PRIVATE
+        ${PROJECT_NAME_UPPER}_SOURCE
+	INTERFACE
+        $<$<BOOL:${WIN32}>:${PROJECT_NAME_UPPER}_NO_LIB>
+        PUBLIC
+	$<$<STREQUAL:$<TARGET_PROPERTY:${PROJECT_NAME},TYPE>,SHARED_LIBRARY>:${PROJECT_NAME_UPPER}_DYN_LINK>
```

Comment:
Maybe this one should be guarded behind a MSVC generator expression

Reply:
It is a CMake generator expression, and always will be resolved in the User's cmake configuration step.  Adding the check Win32 doesn't avoid to have the generator expresion in the generated `fastrtps-...-config.cmake` file. Only when the User's cmake configuration step resolves the generator expression will not put the `-DFASTRTPS_DYN_LINK`, but in linux it doesn't hurt. My opinion is to have the simplier generator expression.